src/main/java/org/sopt/confeti/domain/setlist/application/SetlistService.java

```diff
@@ -118,40 +118,47 @@ public List<Long> createSetLists(Long userId, List<SetlistCreateRequest> request
     }
 
     @Transactional
-    public int addMusics(Long userId, Long setlistId, List<AddSetListMusicRequest> requests) {
+    public int addMusics(Long userId, Long setlistId, List<SetlistAddMusicDTO> requests) {
         Setlist setlist = setlistRepository.findById(setlistId)
                 .orElseThrow(() -> new NotFoundException(ErrorMessage.NOT_FOUND));
 
         if (!setlist.getUser().getId().equals(userId)) {
             throw new UnauthorizedException(ErrorMessage.UNAUTHORIZED);
         }
 
+        List<String> existingIds = setlist.getMusics().stream()
+                .map(SetlistMusic::getTrackId)
+                .toList();
+
         int startOrder = setlist.getMusics().size() + 1;
+        int addedCount = 0;
+
+        for (SetlistAddMusicDTO req : requests) {
+            if(existingIds.contains(req.trackId())) continue;
 
-        for (int i = 0; i < requests.size(); i++) {
-            AddSetListMusicRequest req = requests.get(i);
             SetlistMusic music = SetlistMusic.builder()
                     .trackId(req.trackId())
                     .artistName(req.artistName())
                     .trackName(req.trackName())
                     .artworkUrl(req.artworkUrl())
                     .previewUrl(req.previewUrl())
-                    .orders(startOrder + i)
+                    .orders(startOrder++)
                     .build();
 
             setlist.addMusics(music);
+            addedCount++;
         }
 
-        return requests.size();
+        return addedCount;
     }
 
     public GetSetlistDetailResponse getSetlistDetail(Long userId, Long setlistId) {
         Setlist setlist = setlistRepository.findByIdAndUserId(setlistId, userId)
                 .orElseThrow(() -> new NotFoundException(ErrorMessage.NOT_FOUND));
 
-        List<SetlistMusicResponseDto> musics = setlistMusicRepository.findBySetlist(setlist).stream()
+        List<SetlistMusicResponse> musics = setlistMusicRepository.findBySetlist(setlist).stream()
                 .sorted(Comparator.comparing(SetlistMusic::getOrders))
-                .map(m -> new SetlistMusicResponseDto(
+                .map(m -> new SetlistMusicResponse(
                         m.getId(), m.getTrackId(), m.getArtistName(),
                         m.getTrackName(), m.getArtworkUrl(), m.getPreviewUrl(), m.getOrders()
                 ))
```

Comment:
static 생성자를 사용해서 아래와 같은 방식으로 표현하는 것은 어떠신가요?
매개변수의 순서를 개발 과정에서 헷갈리지 않게 작성할 수 있고, 생성되는 조건을 강제할 수 있어서 더욱 좋을 것 같다는 생각을 합니다!

```suggestion
                .map(SetlistMusicResponse::create)
```

Reply:
앞에서와 비슷하게 정적 팩토리 메서드로 캡슐화 하는 방향으로 수정해볼께요!!!
이후 해당 메서드가 여러 방향으로 변환되는 경우에 Mapper나 Converter를 구현해서 여러 방향의 데이터 변환 책임을 위임하는 것도 좋은 방법이 될 것 같아요!!